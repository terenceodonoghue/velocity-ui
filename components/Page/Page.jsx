/** @jsx jsx */
import { jsx } from '@emotion/core';
import PropTypes from 'prop-types';
import * as S from './Page.styles';

const AppBar = ({ children, ...rest }) => (
  <S.AppBar {...rest}>
    <S.Logo alt="Velocity" src="./logo.svg" />
    <S.Children>{children}</S.Children>
  </S.AppBar>
);

AppBar.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default {
  AppBar,
};
