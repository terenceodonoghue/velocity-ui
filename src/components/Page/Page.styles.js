import styled from '@emotion/styled';

export const AppBar = styled.div`
  display: flex;
  justify-content: space-between;
  border-bottom: solid 1px #e4e8f0;
  padding: 0 112px;
  height: 80px;
  background-color: var(--color-white);
`;

export const Logo = styled.img`
  cursor: pointer;
`;

export const Children = styled.div`
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  margin: 0 -5px;

  & > button {
    margin: 0 5px;
  }

  & > img {
    margin: 0 5px 0 8px;
  }
`;
