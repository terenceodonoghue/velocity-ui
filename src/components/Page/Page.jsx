import React from 'react';
import { arrayOf, oneOfType, node } from 'prop-types';
import * as S from './Page.styles';

const AppBar = ({ children, ...rest }) => (
  <S.AppBar {...rest}>
    <S.Logo alt="Velocity" src="./logo.svg" />
    <S.Children>{children}</S.Children>
  </S.AppBar>
);

AppBar.propTypes = {
  children: oneOfType([arrayOf(node), node]).isRequired,
};

export default {
  AppBar,
};
