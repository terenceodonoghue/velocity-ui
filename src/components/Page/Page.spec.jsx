import React from 'react';
import Page from './Page';

describe('Page', () => {
  describe('AppBar', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(withTheme(<Page.AppBar>AppBar Content</Page.AppBar>))
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
