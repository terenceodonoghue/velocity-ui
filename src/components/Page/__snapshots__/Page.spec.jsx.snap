// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page AppBar matches snapshot 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: solid 1px #e4e8f0;
  padding: 0 112px;
  height: 80px;
  background-color: var(--color-white);
}

.emotion-0 {
  cursor: pointer;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 -5px;
}

.emotion-2 > button {
  margin: 0 5px;
}

.emotion-2 > img {
  margin: 0 5px 0 8px;
}

<div
  className="emotion-4 emotion-5"
>
  <img
    alt="Velocity"
    className="emotion-0 emotion-1"
    src="./logo.svg"
  />
  <div
    className="emotion-2 emotion-3"
  >
    AppBar Content
  </div>
</div>
`;
